app/utils/getFlagCodes.ts

```diff
@@ -0,0 +1,25 @@
+export default function getFlagCode(languages: string[] | undefined) {
+  const flagCodes = {
+    ukr: "UA",
+    ru: "RU",
+    de: "DE",
+    en: "GB",
+    esp: "ES",
+    nl: "NL",
+    nor: "NO",
+    pol: "PL",
+    por: "PT",
+  };
+  const result = [];
```

Comment:
You could directly set `const result = [“DE”]`, skip the logic below and do the `forEach` loop then. If necessary, any double entries could be removed with a `Set`, but from what I get, that’s currently not the case, right?

Reply:
Let me try that. Some listeners have "DE" already, some not, so I think we would need the `Set`

Reply:
Yeah, you can return `[...new Set(whatever)]` :)